### PR TITLE
Unit test case for nimble ansible modules

### PR DESCRIPTION
**Pull request by ar-india**
instruction on how to run the test case is in the readme.

### CODE REVIEW

**Review by ar-india — COMMENTED** [review]
Please review. these are just changes related to unit test case.

File: test/hpe/nimble/unit/hpe_nimble_array_playbook.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        name: "{{ name }}"
+        ctrlr_b_support_ip: "{{ ctrlr_b_support_ip | mandatory}}"
+        ctrlr_a_support_ip: "{{ ctrlr_a_support_ip | mandatory}}"
+        serial: "{{ serial | mandatory}}"
+        nic_list: "{{ nic_list | mandatory}}"
+        pool_name: "{{ pool_name | mandatory}}"
+      register: output
+      failed_when: "'SM_http_not_found' not in output.msg"
+
+    - name: Delete array.Should fail with 'cannot be deleted as it is not present'
```
Comment (suneeth51) [inline test/hpe/nimble/unit/hpe_nimble_array_playbook.yml:57]: nit: Add space
"Delete array. Should fail with 'cannot be deleted as it is not present"

File: test/hpe/nimble/unit/hpe_nimble_encryption_playbook.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        username: "{{ username }}"
+        password: "{{ password }}"
+        name: "default"
+        age: "{{ age | mandatory }}"
+        state: "present"
+        purge_inactive: true
+      tags:
+        - purge_inactive
+
+    - name: group encryption ,will fail as the group name does not exist
```
Comment (suneeth51) [inline test/hpe/nimble/unit/hpe_nimble_encryption_playbook.yml:52]: nit: "group encryption, will fail as the group name does not exist"

File: test/hpe/nimble/unit/hpe_nimble_group_playbook.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        username: "{{ username }}"
+        password: "{{ password }}"
+        name: "{{ name }}"
+        level: 'info'
+        state: "present"
+        test_alert: true
+      tags:
+        - test_alert_group
+
+    - name: Migrate group.Should fail with 'Backup Group Leader Array has not been setup yet'
```
Comment (suneeth51) [inline test/hpe/nimble/unit/hpe_nimble_group_playbook.yml:72]: nit: Add space
"Migrate group. Should fail with 'Backup Group Leader Array has not been setup yet"

File: test/hpe/nimble/unit/hpe_nimble_group_playbook.yml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        host: "{{ host }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        name: "{{ name }}"
+        state: "present"
+        migrate: true
+      register: output
+      failed_when: "'Backup Group Leader Array has not been setup yet' not in output.msg"
+
+    - name: Check migrate group.Should fail with 'Backup Group Leader Array has not been setup yet'
```
Comment (suneeth51) [inline test/hpe/nimble/unit/hpe_nimble_group_playbook.yml:83]: nit:
Check migrate group. Should fail with 'Backup Group Leader Array has not been setup yet'

File: test/hpe/nimble/unit/hpe_nimble_group_playbook.yml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        host: "{{ host }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        name: "{{ name }}"
+        state: "present"
+        check_migrate: true
+      register: output
+      failed_when: "'Backup Group Leader Array has not been setup yet' not in output.msg"
+
+    - name: Merge group.Should fail with 'specified for the argument 'src_passphrase''
```
Comment (suneeth51) [inline test/hpe/nimble/unit/hpe_nimble_group_playbook.yml:94]: nit:
Merge group. Should fail with 'specified for the argument 'src_passphrase'

File: test/hpe/nimble/unit/hpe_nimble_group_playbook.yml [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        host: "{{ host }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        name: "{{ name }}"
+        state: "present"
+        merge: true
+      register: output
+      failed_when: "'SM_invalid_arg_value' not in output.msg"
+
+    - name: Halt group.Should fail with 'SM_invalid_arg_value'
```
Comment (suneeth51) [inline test/hpe/nimble/unit/hpe_nimble_group_playbook.yml:105]: nit:
Halt group. Should fail with 'SM_invalid_arg_value'

**Review by suneeth51 — APPROVED** [review]
lgtm... few minor comments

**Pull request merged**